src/client/src/Client.c

```diff
@@ -1371,6 +1577,14 @@ STATUS freeKinesisVideoClientInternal(PKinesisVideoClient pKinesisVideoClient, S
 
 CleanUp:
 
+    // unlock and free the streamListLock
+    if (locked && IS_VALID_MUTEX_VALUE(pKinesisVideoClient->base.streamListLock)) {

```

Comment:
If it's locked then it had been a valid mutex already

Reply:
Most importantly, you don't need to do this in the CleanUp clause. For the destructors, we don't jump to cleanup while being destructed. So, you can simply freeMutex in it's sequence like others

Reply:
ok I remember running into issues where I had to keep checking for IS_VALID_MUTEX_VALUE during the freeKinesisVideoClientInternal call or else it would segfault, but it's possible I had other bugs then, I'll try with this simpler version.

Reply:
Looks like it. This is an invariant so if it's locked then the lock must have been valid